### ISSUE TITLE
"Save and select" button fails on media image upload screen, when user unselects "reusable".

### ISSUE THREAD

**Issue by kevwalsh** [department-of-veterans-affairs/va.gov-cms#3378]
**Describe the defect**

When adding a new image to the media library in a Story, and making it reusable, the Save and select button fails. 

**To Reproduce**
1. Go to /node/add/news_story
2. Add media 
3. Add a file from your local machine and fill out required fields
4. Uncheck "reusable"
5. Click "save and select"
6. Note that you are returned to the media library browser, but nothing is selected. This is because non-reusable images do not appear on this screen. 

**Expected behavior**

There should be no option that fails to add an image. 

Possible solutions

1. Remove reusability from the content model. It's not in use yet. We can always bring it back later. 
2. Remove "Save and select" button when "Reusable" is checked. 
3. Remove "Save and select" button in contexts where media field is single value. There's no need to be returned to a multi-select form in those contexts.  This has the advantage of a less confusing UI (only one possible action). 
 
Some background: 
* Media was installed with D8.6, and Media updates in 8.8 changed the UI, including the browser and button behavior. 

**Screenshots**

See this video 

https://www.loom.com/share/645115f02bb24fc89094dfc371aa0a20

![Create_Story___VA_gov_CMS.jpg](https://images.zenhubusercontent.com/5d6e57ad330f64311395f08b/a40940d5-033c-40bc-9f40-7db9073e1054)

**Comment by kevwalsh** [comment]
@oksana-c @ndouglas 

Decision reached on Nov 25: "Save and select" should only be visible on media browser screens when Reusable is checked. This allows for editors who want multiselect checkboxes on the ensuing screen to access it when their content is reusable.

**Comment by ndouglas** [comment]
I dug into `#states` trying to figure out what was going wrong, and frankly it defeated me.

The issue is that the dialog loads the form inline and then uses JS to clone the buttons and move them into the dialog, then hides the cloned buttons.  This seems a bit janky to me, but it also breaks `#states`.  A patch was written a year or two ago that copies the `data-drupal-states` attribute from the copied button to the clone, which in theory should allow the clone to observe the state changes, but this stopped working (assuming it ever worked reliably).  The state changes continue to affect the original button and not the clone.  I've played with a few approaches, from forwarding messages to simply deleting the original button, and none of them have worked reliably (and almost surely cause issues elsewhere).  This is a bit of a complicated form operation within a dialog, and I think both the dialog and states components are... ehh.

**TL;DR:** I'm starting to see why these issues have been open and unfixed for a few years.

Anyway, I think I'm going to try to fix this specific case rather than solve the underlying issue with states and dialogs, however much I'd love to do that.  I'm not suitably equipped :rage4:

**Comment by acrollet** [comment]
Just commenting to say that I chatted this over with Nate and I completely support a custom solution in this case. In my (hard-won) experience over the years, the promise of the #states api far exceeds the reality, and I think there's less technical debt in a line or two of our own JS than trying to create and maintain a patch for this issue.

**Comment by oksana-c** [comment]
@ndouglas @acrollet thank you both for this thorough summary

reasonable, feasible, have at it 👍
